src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -543,12 +529,25 @@ internal void AddParentsToNode(GraphNode<RemoteResolveResult> node)
             {
                 lock (_transitiveCentralPackageVersions)
                 {
-                    foreach (var parent in _transitiveCentralPackageVersions[node.Item.Key.Name])
-                    {
-                        node.ParentNodes.Add(parent);
-                    }
+                    List<GraphNode<RemoteResolveResult>> graphNodes = _transitiveCentralPackageVersions[node.Item.Key.Name];
+                    node.ParentNodes.AddRange(graphNodes);
                 }
             }
         }
+
+        private class LibraryDependencyNameComparer : IEqualityComparer<LibraryDependency>
```

Comment:
Is it worth adding a private constructor to these comparers where we only want callers to access the singleton?  I'm wondering if some future dev will accidentally instantiate it instead of using the singleton and we don't have any code analysis to catch it.  Just wondering your thoughts?

Reply:
Good idea. Adding it.